torch/distributed/_tensor/op_schema.py

```diff
@@ -236,6 +237,12 @@ def args_spec(self) -> Tuple[DTensorSpec, ...]:
         """
         # filter out non-relevant values from args schema to get a clean spec list
         # this would mainly be used by sharding propagation rules
+        if self.schema_info is not None and self.schema_info.needs_pytree:
```

Comment:
@wanchaol I had to add these two changes here in `op_schema.py` because even with your `aten.stack` strategy refactor, we still get `needs_redistribute=False` because `args_spec` returns empty tuple, and the sharding propagation logic does not get to compare against the expected input placements.

For `aten.stack`, the `self.args_schema` looks like `([DTensorSpec, DTensorSpec, ...])`, i.e. a singleton tuple whose only element is a `list` of `DTensorSpec` for each stack argument.

I was thinking to use the `needs_pytree` flag as the way to know to peek into the `self.args_schema`, but I am not sure if that is correct.

Reply:
I think this looks reasonable to me!